Enable sklearnex documentation (#587)
intel/scikit-learn-intelex: doc/daal4py/sklearn.rst

to create scikit-learn compatible estimators, transformers, clusterers, etc. powered by oneDAL which
are nearly as efficient as native programs.

Deprecation Notice
-------------------------------

Scikit-learn patching functionality in daal4py was deprecated and moved to a separate

> package - `Intel(R) Extension for Scikit-learn* <https://github.com/intel/scikit-learn-intelex>`_.

Reviewer: ```suggestion
package, `Intel(R) Extension for Scikit-learn* <https://github.com/intel/scikit-learn-intelex>`_.
```

> All future updates for the patching will be available in Intel(R) Extension for Scikit-learn only.

Reviewer: ```suggestion
All future patches will be available only in Intel(R) Extension for Scikit-learn.
```

> Please use the package instead of daal4py for the Scikit-learn acceleration.

Reviewer: ```suggestion
Please use the scikit-learn-intelex package instead of daal4py for the scikit-learn acceleration.
```

plus I'd add a link to scikit-learn-intelex (docs or github)

Author: link is indicated on 15 line

     - pairwise_distance
     - With metric=``cosine`` and ``correlation``.
     - Only dense data is supported.
   * - Other
     - roc_auc_score

>      - Parameters `average`, `sample_weight`, `max_fpr` and `multi_class` are not supported..

Reviewer: ```suggestion
     - Parameters ``average``, ``sample_weight``, ``max_fpr``, and ``multi_class`` are not supported.
```

>      - Parameters ``average``, ``sample_weight``, ``max_fpr`` and ``multi_class`` are not supported..

Reviewer: ```suggestion
     - Parameters ``average``, ``sample_weight``, ``max_fpr``, and ``multi_class`` are not supported.
```
